examples/asr/jasper_aishell_infer.py

```diff
@@ -0,0 +1,186 @@
+    parser.add_argument("--load_dir", type=str, required=True)
+    parser.add_argument("--vocab_file", type=str, required=True)
+    parser.add_argument("--save_logprob", default=None, type=str)
+    parser.add_argument("--lm_path", default=None, type=str)
+    parser.add_argument("--beam_width", default=50, type=int)
+    parser.add_argument("--alpha", default=2.0, type=float)
+    parser.add_argument("--beta", default=1.0, type=float)
+    parser.add_argument("--cutoff_prob", default=0.99, type=float)
+    parser.add_argument("--cutoff_top_n", default=40, type=int)
+
+    args = parser.parse_args()
+    batch_size = args.batch_size
+    load_dir = args.load_dir
+
+    if args.local_rank is not None:
+        if args.lm_path:
+            raise NotImplementedError(
+                "Beam search decoder with LM does not currently support "
+                "evaluation on multi-gpu.")
+        device = nemo.core.DeviceType.AllGpu
+    else:
+        device = nemo.core.DeviceType.GPU
+
+    # Instantiate Neural Factory with supported backend
+    neural_factory = nemo.core.NeuralModuleFactory(
+        backend=nemo.core.Backend.PyTorch,
+        local_rank=args.local_rank,
+        optimization_level=nemo.core.Optimization.mxprO1,
+        placement=device)
+    logger = neural_factory.logger
+
+    if args.local_rank is not None:
+        logger.info('Doing ALL GPU')
+
+    yaml = YAML(typ="safe")
+    with open(args.model_config) as f:
+        jasper_params = yaml.load(f)
+
+    vocab = load_vocab(args.vocab_file)
+
+    sample_rate = jasper_params['sample_rate']
+
+    eval_datasets = args.eval_datasets
+
+    eval_dl_params = copy.deepcopy(jasper_params["AudioToTextDataLayer"])
+    eval_dl_params.update(jasper_params["AudioToTextDataLayer"]["eval"])
+    eval_dl_params["normalize_transcripts"] = False
+    del eval_dl_params["train"]
+    del eval_dl_params["eval"]
+    data_layer = nemo_asr.AudioToTextDataLayer(
+        manifest_filepath=eval_datasets,
+        sample_rate=sample_rate,
+        labels=vocab,
+        batch_size=batch_size,
+        **eval_dl_params)
+
+    n = len(data_layer)
+    logger.info('Evaluating {0} examples'.format(n))
+
+    data_preprocessor = nemo_asr.AudioPreprocessing(
+        sample_rate=sample_rate,
+        **jasper_params["AudioPreprocessing"])
+    jasper_encoder = nemo_asr.JasperEncoder(
+        feat_in=jasper_params["AudioPreprocessing"]["features"],
+        **jasper_params["JasperEncoder"])
+    jasper_decoder = nemo_asr.JasperDecoderForCTC(
+        feat_in=jasper_params["JasperEncoder"]["jasper"][-1]["filters"],
+        num_classes=len(vocab))
+    greedy_decoder = nemo_asr.GreedyCTCDecoder()
+
+    if args.lm_path:
+        beam_width = args.beam_width
+        alpha = args.alpha
+        beta = args.beta
+        cutoff_prob = args.cutoff_prob
+        cutoff_top_n = args.cutoff_top_n
+        beam_search_with_lm = nemo_asr.BeamSearchDecoderWithLM(
```

Comment:
Can you add documentation to `BeamSearchDecoderWithLM` to state what cutoff_prob, and cutoff_top_n mean?
The default value of 1.0 for cutoff_prob seems to suggest that you are cutting things off, which is not consistent with prior implementations.

Reply:
The default value 1.0 means cutoff nothing. You may refer to baidu's implementation.
Once you've installed the decoder, you will find the documentation about this parameters under decoders/ctc_decoder.py.